statistics-production/api-data-standards.qmd

```diff
@@ -0,0 +1,191 @@
+
+Whilst the EES data screener tests for a significant base level of data quality and consistency, there are some 
+additional criteria that are either too awkward to test for rigorously using the screener or are tested for but
+returned as warnings. Data intended for the EES API must pass all the base level screener tests, plus a number
+that only return warnings, plus manual inspection by the platform gatekeepers. These are primarily:
+
+- Strict tidy data structures - i.e. appropriate use of filters and indicators.
+- Standardised filter col_names and items consistent with the harmonised standards.
+- Standardised indicator col_names meeting the naming standards.
+- Character limits for col_names and filter items.
+
+Examples of these that do and don't meet the API data standards are provided in the following sections.
+
+## Tidy data structure
+
+The key thing on tidy data structure is to avoid filter items being included within indicator col_names. Where 
+you have collections of related terms appearing in indicator names (e.g. male, female, total), then these 
+should be translated into a filter column, with the data being pivoted.
+
+### Example 1 - Attainment rates and scores
+
+#### Example of bad practice
+
+The following would not be accepted for publication via the API.
+
+::: {.table-responsive}
+
+| count_pupils_grade9to5 | count_pupils_grade9to4 | count_pupils_grade9to1 | percent_pupils_grade9to5 | percent_pupils_grade9to4 | percent_pupils_grade9to1 | progress8_score_male | progress8_score_female | progress8_score |attainment8_score_male | attainment8_score_female | attainment8_score |
+|------------------------|------------------------|------------------------|--------------------------|--------------------------|-------------------------|----------------------|------------------------|-----------------|----------------------|------------------------|-----------------|
+| 30                     | 40                     | 50                     | 60                       | 80                       | 100                      | 0.2                  | 0.21                   | 0.21            |0.09                  | 0.08                   | 0.10            |
+
+: Attainment grade rates and scores in non-tidy format
+
+:::
+
+#### Example of good practice
+
+The following would be accepted for publication via the API. In this case, splitting the data into separate data files is required in order to create tidy data structures.
+
+::: {.table-responsive}
+
+|attainment_metric   | count_pupils | percent_pupils |
+|--------------------|--------------|----------------|
+| Grades 9-5         | 30           | 60             |          
+| Grades 9-4         | 40           | 80             |         
+| Grades 9-1         | 50           | 100            |                  
+
+: Attainment grade rates in tidy format
+
+:::
+
+::: {.table-responsive}
+
+| sex    | attainment_metric | score_average  |
```

Comment:
Nice example as I can see debate over this - what about `average_progress_8` / `average_attainment_8` ?

Reply:
Potentially, but my main issue with putting the name of the attainment metric in an indicator name is that it seems pretty likely that at some point, a minister would ask for a review of attainment metrics which could then say e.g. "Progress 8" isn't what we want to measure, we're going to invent a new scoring methodology and call it "Education gains" or something, at which point we've got whole new set of indicators in our attainment data set. By having the metric descriptor held in a filter, it feels more robust to me.

Reply:
That's a fair point, I think my issue lies with where the line is between molten and tidy data. We have set a line elsewhere that metric generally isn't an appropriate filter (e.g. count / percent shouldn't be filter options).

I think whole new sets of indicators are probably expected in attainment data, so that doesn't bother me. 

I'm not against taking this line / approach, though would like be more certain on first, are there any non-attainment examples of similar kinds of policy / topic area indicators (I don't know, maybe count_fines_issued?) that could be helpful to have as examples as well? SEN might be an interesting one to show actually if they do anything reporting on numbers of plans instead of counting pupils?

The more I think about it the more I'm happy with your line on this, I think I just want some extra examples in so it's really clear what the line is between a filter and indicator.

Reply:
Not entirely the point of your comment, but just to pick up on the specific detail, we could change that to `attainment_framework` or [`accountability_measure`](https://www.gov.uk/government/publications/progress-8-school-performance-measure) if either of those sit better than using the term metric?

Reply:
Attendance could be a useful one as I've suggested `session_count` / `session_percent` and `pupil_count` / `pupil_percent` for their data.

Reply:
Yeah either of those attainment ones could work - tbh probably one to agree with the different key stage teams in Mark's unit (thinking of the data dictionary we will have to settle on one they're all happy with at some point)

Reply:
Yeah, totally. These ones are still very much on the table for discussion (and this might push some of that discussion to happen sooner).